src/test/scala/ai/privado/languageEngine/java/tagger/JavaS3TaggerTest.scala

```diff
@@ -27,72 +27,108 @@ import ai.privado.cache.{AppCache, S3DatabaseDetailsCache}
+      false,
+      "",
+      Map(),
+      NodeType.REGULAR,
+      "",
+      CatLevelOne.SINKS,
+      "",
+      Language.JAVA,
+      Array()
+    ),
+    RuleInfo(
+      "Storages.AmazonS3.Write",
+      "Amazon S3",
+      "Storage",
+      FilterProperty.METHOD_FULL_NAME,
+      Array(),
+      List(".*PutObjectRequest.*"),
+      false,
+      "",
+      Map(),
+      NodeType.REGULAR,
+      "",
+      CatLevelOne.SINKS,
+      "",
+      Language.JAVA,
+      Array()
+    )
+  )
 
   "Java code reading and writing from S3 bucket" should {
+
+    val ruleCache = RuleCache().setRule(
+      RuleInfoTestData.rule
+        .copy(sources = RuleInfoTestData.sourceRule, sinks = sinkRule)
+    )
+
+    val cpg = code("""
+        |import software.amazon.awssdk.core.sync.RequestBody;
+        |import software.amazon.awssdk.services.s3.S3Client;
+        |import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+        |import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+        |
+        |public class AWS {
+        |    public static void main(String[] args) {
+        |        S3Client s3Client = S3Client.builder().build();
+        |        String objectKey = "your-object-key";
+        |        String filePath = "path/to/your/file.txt";
+        |
+        |        PutObjectRequest request = PutObjectRequest.builder()
+        |            .bucket("my-write-bucket")
+        |            .key(objectKey)
+        |            .build();
+        |
+        |        GetObjectRequest request2 = GetObjectRequest.builder()
+        |                .key(objectKey)
+        |                .bucket("my-read-bucket")
+        |                .build();
+        |
+        |        ResponseBytes<GetObjectResponse> objectBytes = s3Client.getObjectAsBytes(request2);
+        |            byte[] data = objectBytes.asByteArray();
+        |
+        |            // Write the data to a local file.
+        |            File myFile = new File(filePath);
+        |            OutputStream os = new FileOutputStream(myFile);
+        |            os.write(data);
+        |            System.out.println("Successfully obtained bytes from an S3 object");
+        |            os.close();
+        |
+        |        s3Client.putObject(request, RequestBody.fromFile(new File(filePath));
+        |        s3Client.close();
+        |    }
+        |}
+        |""".stripMargin)
+      .withRuleCache(ruleCache)
+
     "have bucket name" in {
-      val sinkExporter =
-        new SinkExporter(
-          cpg,
-          ruleCache,
-          privadoInput,
-          None,
-          s3DatabaseDetailsCache,
-          appCache = new AppCache(),
-          databaseDetailsCache = databaseDetailsCache
-        )
-      sinkExporter.getSinks.map(_.databaseDetails.dbName) shouldBe List("my-write-bucket", "my-read-bucket")
+      val outputMap = cpg.getPrivadoJson()
+      val sinks = outputMap(Constants.sinks)
```

Comment:
Thanks for trying to fix the legacy test case, One change here similar to how you have created a trait and got the SinkProcessing at https://github.com/Privado-Inc/privado-core/pull/1240/files#diff-db0257a7815ec7fd3a87ff7ce71e9484408a9f6682c298928935f9076dc68672R10 Create another method in the same trait to `getSinks` and use it below

Reply:
done.